zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/variable/VariableDocumentUpdateProcessor.java

```diff
@@ -67,7 +67,8 @@ public void processRecord(final TypedRecord<VariableDocumentRecord> record) {
         new AuthorizationRequest(
                 record,
                 AuthorizationResourceType.PROCESS_DEFINITION,
-                PermissionType.UPDATE_PROCESS_INSTANCE)
+                PermissionType.UPDATE_PROCESS_INSTANCE,
+                value.getTenantId())
```

Comment:
❌ Tenant is not provided by users for this command. Here we should also make sure to get the tenant from the element instance.